src/neuroconv/utils/str_utils.py

```diff
@@ -0,0 +1,47 @@
+def human_readable_size(size_bytes: int, binary=True) -> str:
+    """
+    Convert a file size given in bytes to a human-readable format.
+
+    Parameters
+    ----------
+    size_bytes : int
+        The size in bytes.
+    binary : bool, optional
+        If True, use binary prefixes (KiB, MiB, etc.). If False, use SI prefixes
+
+    Returns
+    -------
+    str
+        A human-readable string representation of the size.
+
+    Examples
+    --------
+    >>> human_readable_size(123)
+    '123 B'
+    >>> human_readable_size(1234)
+    '1.21 KB'
+    >>> human_readable_size(123456789)
+    '117.74 MB'
+    """
+    # Check if size is negative
+    if size_bytes < 0:
+        raise ValueError("Size must be non-negative")
+
+    # Define the suffixes for each size unit
+    suffixes = ["", "K", "M", "G", "T", "P", "E", "Z", "Y"]
+    i = 0
+    # Convert bytes to a float for division
+    size_float = float(size_bytes)
+
+    base = 1000 if not binary else 1024
+
+    # Find the appropriate unit
+    while size_float >= base and i < len(suffixes) - 1:
```

Comment:
I thought there was a non-iterate one-shot calculation you had using a log of the binary (or decimal) base?